wire-ios/Wire-iOS/Sources/SwiftUI Helpers/HapticFeedbackModifier.swift

```diff
@@ -0,0 +1,50 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+
+func withFeedback(
+    _ style: UINotificationFeedbackGenerator.FeedbackType,
+    _ action: @escaping () -> Void
+) -> () -> Void {
+    { () in
+        let feedback = UINotificationFeedbackGenerator()
+        feedback.prepare()
+        feedback.notificationOccurred(style)
+        action()
+    }
+}
+
+struct HapticTapGestureViewModifier: ViewModifier {
+    var style: UINotificationFeedbackGenerator.FeedbackType
+    var action: () -> Void
+
+    func body(content: Content) -> some View {
+        content.onTapGesture(perform: withFeedback(self.style, self.action))
+    }
+}
+
+extension SwiftUI.Button {
+    init(
+        _ style: UINotificationFeedbackGenerator.FeedbackType,
+        action: @escaping () -> Void,
+        @ViewBuilder label: () -> Label
+    ) {
+        self.init(action: withFeedback(style, action), label: label)
+    }
```

Comment:
Why create an initializer instead of using the modifier?

Reply:
Same thing applies here: https://github.com/wireapp/wire-ios-mono/pull/167#discussion_r1170047560